src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,18 +32,35 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
+        this.splittedAddressArray = splitAddressIntoArray(trimmedAddress);
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        if (!isValidAddress(this.splittedAddressArray)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        this.block = new Block (splittedAddressArray[0]);
+        this.street = new Street (splittedAddressArray[1]);
+        this.unit = new Unit (splittedAddressArray[2]);
+        this.postalCode = new PostalCode (splittedAddressArray[3]);
+    }
+
+
+    /**
+     * Split input address string into a String array, consisting of the block, street, unit and postal code.
+     * 
+     * @param address the input address String
+     * @return array of String
+     */
+    private String[] splitAddressIntoArray(String address) {
+        String[] addressArray = address.split(", ");
+        return addressArray;
     }
 
     /**
      * Returns true if a given string is a valid person email.
      */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    public static boolean isValidAddress(String[] test) {
+        return test.length==4;
```

Comment:
toString() method should be using the new classes, instead of returning the 'value' user entered.